node/src/syncer.rs

```diff
@@ -712,63 +806,76 @@ mod tests {
+        assert_syncing(&syncer, &store, &[1500..=1500], 1500).await;
 
-        let (syncer, store, mut p2p_mock) = initialized_syncer(headers[25].clone()).await;
-        assert_syncing(&syncer, &store, &[26..=26], 26).await;
+        // Syncer is syncing backwards from the network head (batch 1)
+        handle_session_batch(&mut p2p_mock, &headers, 988..=1499, true).await;
+        assert_syncing(&syncer, &store, &[988..=1500], 1500).await;
 
-        // Syncer will sync all headers up to the head
-        handle_session_batch(&mut p2p_mock, &headers, 1..=25, true).await;
-        assert_syncing(&syncer, &store, &[1..=26], 26).await;
+        // Syncer is syncing backwards from the network head (batch 2)
+        handle_session_batch(&mut p2p_mock, &headers, 476..=987, true).await;
+        assert_syncing(&syncer, &store, &[476..=1500], 1500).await;
+
+        // New HEAD was received by HeaderSub (height 1501)
+        let header1501 = gen.next();
+        p2p_mock.announce_new_head(header1501.clone());
+        // Height 1501 is adjacent to the last header of Store, so it is appended
+        // immediately
+        assert_syncing(&syncer, &store, &[476..=1501], 1501).await;
+
+        // Syncer is syncing backwards from the network head (batch 3, partial)
+        handle_session_batch(&mut p2p_mock, &headers, 1..=475, true).await;
+        assert_syncing(&syncer, &store, &[1..=1501], 1501).await;
 
         // Syncer is fulling synced and awaiting for events
         p2p_mock.expect_no_cmd().await;
 
-        // New HEAD was received by HeaderSub (height 27)
-        let header27 = gen.next();
-        p2p_mock.announce_new_head(header27.clone());
-        // Height 27 is adjacent to the last header of Store, so it is appended
-        // immediately
-        assert_syncing(&syncer, &store, &[1..=27], 27).await;
+        // New HEAD was received by HeaderSub (height 1502), it should be appended immediately
+        let header1502 = gen.next();
+        p2p_mock.announce_new_head(header1502.clone());
+        assert_syncing(&syncer, &store, &[1..=1502], 1502).await;
         p2p_mock.expect_no_cmd().await;
 
-        // New HEAD was received by HeaderSub (height 30), it should NOT be appended
-        let header_28_30 = gen.next_many(3);
-        p2p_mock.announce_new_head(header_28_30[2].clone());
-        assert_syncing(&syncer, &store, &[1..=27], 30).await;
+        // New HEAD was received by HeaderSub (height 1506), it should NOT be appended
+        let headers_1503_1506 = gen.next_many(3);
+        p2p_mock.announce_new_head(headers_1503_1506[2].clone());
+        assert_syncing(&syncer, &store, &[1..=1502], 1505).await;
 
         // New HEAD is not adjacent to store, so Syncer requests a range
-        handle_session_batch(&mut p2p_mock, &header_28_30, 28..=30, true).await;
-        assert_syncing(&syncer, &store, &[1..=30], 30).await;
+        handle_session_batch(&mut p2p_mock, &headers_1503_1506, 1503..=1505, true).await;
```

Comment:
Because of previous suggestion

```suggestion
        handle_session_batch(&mut p2p_mock, &headers_1503_1505, 1503..=1505, true).await;
```